### PR TITLE
Update kedro pipeline create to use new /conf file structure

### LINKED ISSUE #2752: Update kedro pipeline create to use new /conf file structure
## Description
Currently, the `kedro pipeline create` command creates a new /parameters folder and adds a {pipeline-name}.yml inside (or just adds the file to the parameters folder if it aready exists. 

## Context
To make this more consistent and so the global parameters.yml doesn't just randomly sit outside the directories, we want to maintian all the pipeline specific config files directly in the /conf with a specific naming convention. 

Before
```
├── conf
    └── base
        └── parameters         
	    ├── data_science.yml     #blank files
        ├── catalog.yml              #blank files   
	└── parameters.yml           #blank files
```

After
```
├── conf
    └── base
	├── catalog_data_science.yml          #blank files
	└── parameters_data_science.yml       #blank files
        ├── catalog.yml          #blank files   
	└── parameters.yml       #blank files
```

## Possible Implementation

calling the updated `kedro pipeline create` should: 
1. add `parameters_{pipeline_name}.yml` to `/conf`
2. add `catalog_{pipeline_name}.yml` to `/conf`
3. add a pipeline diretory called {pipeline_name} to the `/pipelines` with empty `nodes.py` and `pipeline.py` files 

Currently, it also adds a `/tests` directory, but we want to remove this from the command because of work on #2149

### PR DESCRIPTION
**Pull request by DimedS**
Fixes #2752

## Description
This Pull Request (PR) refines the Kedro pipeline create command. Instead of nesting a new "{pipeline-name}.yml" file in a "/parameters" folder, we now directly create a "parameters_{pipeline-name}.yml" file in the conf folder. This change simplifies the directory structure and improves user accessibility.

## Development notes
I moved the {pipeline-name}.yml /template file from the parameters folder to the conf folder and added "parameters_" at the start of the filename.

This change meant I had to tweak the delete_pipeline function in cli/pipeline.py so it knew where to find the right files to delete.

After these changes, a lot of tests stopped working because they were set up for the old folders. I fixed these tests in test_pipeline.py.

I also had to make changes for the microPackaging. I added two new search templates to 'configs_to_package' in cli/micropkg.py, but I kept the two old ones that use the "parameters" folder, so we don't break anything for users who use the old structure.

I fixed the tests for these changes in test_micropkg_package.py and test_micropkg_pull.py.

While fixing the tests, I noticed I was repeating a lot of the same fixes. We should look at redesigning the tests later to make things more consistent.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by merelcht — COMMENTED** [review]
Congrats on your first PR @DimedS 😄 ⭐ 

The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

**Review by noklam — COMMENTED** [review]
Leave some comments. But more importantly I notice `micropkg` is completely broken yet our `e2e-test` have no problem about it. I am not an expert on `micropkg` so I need some help.

See the directory here when I do `kedro micropkg pacakge pipelines.nok` and `tar -xf` in `main` branch

```
├── MANIFEST.in
├── nok
│   ├── config
│   │   └── parameters_nok.yml
│   ├── __init__.py
│   ├── nodes.py
│   ├── pipeline.py
│   └── README.md
├── nok.egg-info
│   ├── dependency_links.txt
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   └── top_level.txt
├── PKG-INFO
├── pyproject.toml
├── setup.cfg
└── tests
    ├── __init__.py
    └── test_pipeline.py
```

I expect this should be `conf` instead of `config`, but it seems that this has been like this for long time, do we expect a micropkg package is `kedro runable`?. @merelcht

**Comment by noklam** [comment]
> Congrats on your first PR @DimedS 😄 ⭐
> 
> The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

Should this be a e2e-test?

**Comment by DimedS** [comment]
> Congrats on your first PR @DimedS 😄 ⭐
> 
> The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

Merel, thank you for your insights, I fully agree! I've added the requested feature, along with the corresponding unit tests. Additionally, I've updated the Release.MD file and re-requested a review.

File: kedro/framework/cli/micropkg.py [hunk 1]
```diff
@@ -613,9 +613,16 @@ def _package_micropkg(
...
+    # because parameters had been moved from foldername to yml filename
     configs_to_package = _find_config_files(
         package_conf,
-        [f"parameters*/**/{micropkg_name}.yml", f"parameters*/**/{micropkg_name}/**/*"],
+        [
+            f"**/parameters_{micropkg_name}.yml",
+            f"**/{micropkg_name}/**/*",
+            f"parameters*/**/{micropkg_name}.yml",
+            f"parameters*/**/{micropkg_name}/**/*",
+        ],
```
[the remark this answers is not in the document]
Comment (DimedS) [reply kedro/framework/cli/micropkg.py:625]: Nok hi, thank you for your review. In my opinion, currently it's not configurable. However, as you've suggested, it seems like a good idea to refactor the micropkg module code in a separate ticket. At this time, I've simply updated the default 'parameters' structure while maintaining backward compatibility.

File: kedro/framework/cli/pipeline.py [hunk 2]
```diff
@@ -140,10 +140,19 @@ def delete_pipeline(
...
+    ]
+
+    # Since we remove nesting in 'parameters' and 'catalog' folders,
+    # we want to save also the old project's structure for the 'pipeline delete' command for backward compatibility
+    files_to_delete += [
         pipeline_artifacts.pipeline_conf / confdir / f"{name}.yml"
         for confdir in ("parameters", "catalog")
         if (pipeline_artifacts.pipeline_conf / confdir / f"{name}.yml").is_file()
     ]
+
```
Comment (noklam) [inline kedro/framework/cli/pipeline.py:155]: Can it be refactored as looping the two patterns? There are many repeat code here, the only difference is `f"{confdir}_{name}.yml"` and  `confdir / f"{name}.yml"`

Comment (DimedS) [reply kedro/framework/cli/pipeline.py:150]: Thanks for catching that, Nok. Fixed it  👍

**Review by merelcht — APPROVED** [review]
Thanks @DimedS, great work! ⭐

**Review by noklam — APPROVED** [review]
Thanks for addressing the comment. I think this is good enough to be merged. What's your thought about the e2e test, do you still think it is worth it?

**Comment by noklam** [comment]
I think this is ready and I have approved it already.

**Pull request merged**